### PR TITLE
sql: support MATERIALIZED modifier for CREATE SOURCES

### PR DESCRIPTION
**Pull request by petrosagg**
If specified all sub-sources will become materialized.

### CODE REVIEW

File: src/sql-parser/tests/testdata/ddl [hunk 1]
```diff
@@ -550,14 +550,21 @@ CREATE SOURCES FROM POSTGRES HOST 'host=kanto user=ash password=teamrocket dbnam
...
+
+parse-statement
+CREATE MATERIALIZED SOURCES FROM POSTGRES HOST 'host=kanto user=ash password=teamrocket dbname=pokemon' PUBLICATION 'red' NAMESPACE 'generation1' TABLES ('random.psychic' as "public.psychic" (pokedex_id int NOT NULL, evolution int), 'another_table' as "another_one")
+----
+CREATE SOURCES FROM POSTGRES HOST 'host=kanto user=ash password=teamrocket dbname=pokemon' PUBLICATION 'red' NAMESPACE 'generation1' TABLES ('random.psychic' AS "public.psychic" (pokedex_id int4 NOT NULL, evolution int4), 'another_table' AS another_one)
```
Comment (JLDLaughlin) [inline src/sql-parser/tests/testdata/ddl:565]: This is missing the `MATERIALIZED` keyword, which you can add in [here](https://github.com/MaterializeInc/materialize/blob/12c842aacdfc84b4f7160a95eb200db1b3dd1c04/src/sql-parser/src/ast/defs/statement.rs#L409)!

**Review by JLDLaughlin — APPROVED** [review]
🎉

File: src/sql-parser/tests/testdata/ddl (hunk 1 above)
Comment (petrosagg) [reply src/sql-parser/tests/testdata/ddl:565]: good catch! I need to be more careful with this. shouldn't the test fail if what goes in the parser is different than what comes out?

Comment (JLDLaughlin) [reply src/sql-parser/tests/testdata/ddl]: the `sql-parser` tests will only compare against the provided output! which makes small things like this easy to miss, for sure.

**Pull request merged**